js/scripts/util.js

```diff
@@ -133,4 +140,86 @@ const checkBlockCollisions = (entity) => {
     entity.boundingBox = new BoundingBox(entity.pos, entity.constructor.SCALED_SIZE);
 
     return collisions;
+};
+
+// The following is necessary because we must change the listeners for different modes (right now, gameplay and dialog).
+/** Contains all functions called as event handlers. */
+const EVENT_HANDLERS = {
+    gameplayMouseDown: (e) => {
+        GAME.user.firing = false;
+        GAME.user.aiming = true;
+    },
+    gameplayMouseUp: (e) => {
+        GAME.user.aiming = false;
+        GAME.user.firing = true;
+    },
+    gameplayMouseMove: (e) => {
+        const rect = CANVAS.getBoundingClientRect();
```

Comment:
What does CANVAS.getBoundingClientRect do? is that the bounding box for Chad? Can you explain?

Reply:
Ask @NathanHinthorne

Reply:
There's a lot of details in getBoundingClientRect(). It's based on stuff related to the canvas the user sees. The function helps with the conversion process of mouse-to-canvas coordinates, ensuring the coordinates are scaled properly based on the user's screen size. I ran into a lot of bugs and decided to use this function to solve the problem.

Here's some more info on it:
https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect

Reply:
Thank you!